sw/otbn/crypto/div.s

```diff
@@ -0,0 +1,369 @@
+/* Copyright lowRISC contributors. */
+/* Licensed under the Apache License, Version 2.0, see LICENSE for details. */
+/* SPDX-License-Identifier: Apache-2.0 */
+
+/**
+ * Shift a bignum one bit to the right.
+ *
+ * Returns a' = lsb(b) || a >> 1.
+ *
+ * This is a customized helper function for `div`. Modifies the input in-place.
+ * This routine runs in constant time.
```

Comment:
General question for my understanding, does the constant time checker model the instructions that take 1 cycle versus 2 cycles? 

Does it run on top of the instruction simulator?

Reply:
It's simpler than that -- it's a static check that builds an information-flow graph and checks if any control flow depends on values that may be secret. So it can have false negatives (e.g. two branches with the same number of cycles), but shouldn't have false positives because if it passes then you should always execute the exact same instruction sequence regardless of secret inputs.

Reply:
ah sounds good. I was thinking about cases where some instructions take 2 cycles instead of 1, and unsure if there are any circumstances where the control flow (in other programs, not this PR), where a code snippet may end up taking an un-intended extra cycle. 

I think the likelihood of this is low, but thought it is something the constant time checker could incorporate.

Reply:
Hmm, I know there are some that take 2 cycles always (control flow, load/store). And then there's `RND` reads, which take however long the entropy complex needs (but it's never secret-dependent). Are there any others? There are pseudoinstructions that expand to 1 or 2 instructions depending on e.g. the immediate value, but these are expanded in the assembler stage and the constant-time checker consumes the binary.